app/models/manageiq/providers/amazon/inventory/collector/target_collection.rb

```diff
@@ -16,6 +16,20 @@ def name_references(collection)
     target.manager_refs_by_association.try(:[], collection).try(:[], :name).try(:to_a).try(:compact) || []
   end
 
+  def flavors
+    return @flavors_hashes if @flavors_hashes
+
+    all_instance_types = ManageIQ::Providers::Amazon::InstanceTypes.instance_types
+
+    # Only include flavors which are referenced by the targeted instances
+    targeted_instance_types = instances.collect { |instance| instance["instance_type"] }.uniq.compact
+
+    instance_types = all_instance_types.values_at(*targeted_instance_types)
+    instance_types << all_instance_types["unknown"] if instance_types.any?(nil)
+
+    @flavors_hashes = instance_types.compact
+  end
```

Comment:
Not for this PR, but I wonder if it's cleaner to just have the instance_types Hash return "unknown" as a default value.  Then I think this code simplifies to:

```suggestion
  def flavors
    return @flavors_hashes if @flavors_hashes

    # Only include flavors which are referenced by the targeted instances
    targeted_instance_types = instances.collect { |instance| instance["instance_type"] }

    @flavors_hashes = ManageIQ::Providers::Amazon::InstanceTypes.instance_types.values_at(*targeted_instance_types).uniq
  end
```

Reply:
Done